Add seeding for Record, Certificate, and Challenge

cychu42 opened the pull request:
Closes: #143 
Let me know if you think more example data should be included, as well as whether the examples need changes.

## Changes
- Code for seeding Record, Certificate, and Challenge tables in `prisma/seed.ts`.
- Code for cleaning all existing data in the database in `prisma/seed.ts`.

## Testing
1. Set up the project normally to see if seeding works
or
2. run `npx prisma db seed` in a working copy of the project to test it

humphd commented on prisma/seed.ts:
```diff
@@ -23,6 +25,118 @@ async function seed() {
+    data: [
+      // A record
+      {
+        id: 1,
```
Based on https://github.com/Seneca-CDOT/starchart/blob/main/prisma/schema.prisma#L23, you shouldn't need to provide an `id`, it should automatically assign a new one.

humphd replied:
Based on the call today, sounds like you want this for foreign key.  Resolving.

humphd commented on prisma/seed.ts:
```diff
@@ -23,6 +25,118 @@ async function seed() {
+        name: 'OSD700',
+        type: `A`,
+        value: '172.5.251.0',
+        description: 'For final porject.',
```
Typo in `project`

cychu42 replied:
Ah, thanks.

humphd commented on prisma/seed.ts:
```diff
@@ -23,6 +25,118 @@ async function seed() {
+      {
+        id: 1,
+        username: username,
+        name: 'OSD700',
```
Prefer lowercase for all `name` values`

humphd commented on prisma/seed.ts:
```diff
@@ -23,6 +25,118 @@ async function seed() {
+      // A record
+      {
+        id: 1,
+        username: username,
```
When the key/value are the same, list it once: `username,`

cychu42 replied:
I see. Thank you.

humphd commented on prisma/seed.ts:
```diff
@@ -23,6 +25,118 @@ async function seed() {
+
+  await prisma.certificate.create({
+    data: {
+      id: 1,
```
Same comments here.  Let Prisma do the automatic fields for you.

SerpentBytes commented on prisma/seed.ts:
```diff
@@ -23,6 +25,118 @@ async function seed() {
+      certificate:
+        '-----BEGIN CERTIFICATE-----ApfFCv0O65TYkp5jEWSlU8PhKYD43nXA=-----END CERTIFICATE-----',
+      orderUrl: `orderURL.example.com`,
+      privkey:
```
Maybe follow camelCase convention here to be consistent with the rest of the code.

cychu42 replied:
OK, done.

cychu42 commented:
Thank you for the feedbacks. I did a number of changes:
- I would like to keep the specified `id`s to avoid situations where there’s is an issue referencing them for foreign keys. Otherwise. auto incrementing will just keep incrementing the number every time someone runs the seeding command.
- For dates, I changed dates of creation to be now. Expiration dates are a certain time from now. 6 months for records, and 90 days for certificates.
- The challenge part is removed as Denes suggested.
- `orderUrl` is also changed as suggested.

SerpentBytes commented:
@cychu42 is privkey supposed to be privKey?

humphd reviewed (changes requested):
Looks good, other than the `privKey` vs. `privkey` change @SerpentBytes wants.

cychu42 commented:
> @cychu42 is privkey supposed to be privKey?

It's supposed to be `privkey`, as it's what's in the schema. However, I see that it should probably be `privKey` or even `privateKey`. I originally took it from the [Data Architecture](https://github.com/Seneca-CDOT/starchart/wiki/Data-Architecture) page when making the schema.
I think the later (`privateKey`) makes more sense as it will be clearer to new readers.

If we are going to change it, I can add it as a suggestion under another PR to change the schema, like #153. Just to keep scope of issues separate.
